### PR TITLE
Support augmented documents and search queries in chat (non-streaming)

### PR DESCRIPTION
**Pull request by BeatrixCohere**
Closes PTS-4548

Adds functionality to chat to support augmented documents and generating search queries in chat.

### CODE REVIEW

File: cohere/client.py [hunk 1]
```diff
@@ -269,6 +276,16 @@ def chat(
                 >>>     return_prompt=True)
                 >>> print(res.text)
                 >>> print(res.prompt)
+            Query generation example:
+                >>> res = co.chat(query="What are the tallest penguins?", mode="search_query_generation")
```
Comment (mkozakov) [inline cohere/client.py:280]: i think we want to use the enum values for mode, not strings right? this applies to the tests as well

File: cohere/responses/chat.py [hunk 2]
```diff
@@ -1,25 +1,31 @@
 import json
+from enum import Enum
 from typing import Any, Dict, Generator, List, NamedTuple, Optional
 
 import requests
 
 from cohere.responses.base import CohereObject
 
+Mode = Enum("Mode", ["chat", "search_query_generation", "augmented_generation"])
```
Comment (mkozakov) [inline cohere/responses/chat.py:9]: i think enum values are normally all caps

**Review by mkozakov — APPROVED** [review]

File: cohere/client.py (hunk 1 above)
Comment (sanderland) [reply cohere/client.py:280]: that can be somewhat annoying to use though, having to import the enum object from somewhere, and other endpoints just take strings.

File: cohere/responses/chat.py (hunk 2 above)
Comment (sanderland) [reply cohere/responses/chat.py:9]: Have never really used Enum, but after testing around a bit, this doesn't quite seem what you want?
```
Mode = Enum("Mode", ["chat", "search_query_generation", "augmented_generation"])
Mode.chat=='chat' # False
Mode.chat.value == 1 # True
```

File: cohere/client.py (hunk 1 above)
Comment (BeatrixCohere) [reply cohere/client.py:280]: This should work with both importing mode and with the string

**Comment by mkozakov** [comment]
Do you know why tests are still failing?

**Review by mkozakov — APPROVED** [review]

**Pull request merged**